### PR TITLE
Remove dead code in RzUtil alloc

### PR DESCRIPTION
**Pull request by XVilka**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [x] I've documented or updated the documentation of every function and struct this PR changes. If not so I've explained why.
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

All those `malloc()`, `free()`, etc wrappers are unnecessary and generally unused. Removed them and few places where they were used.

There is no good cross-platform aligned malloc function, thus I left `rz_alloc_aligned()` and used `posix_memalign()` or `_aligned_malloc()` (Windows) where they exist (detected via Meson).

**Test plan**

CI is green

### CODE REVIEW

File: librz/util/alloc.c [hunk 1]
```diff
@@ -4,72 +4,33 @@
...
-	return _r_realloc(p, sz);
-}
-
-RZ_API void rz_free(void *p) {
-	return _r_free(p);
-}
-#endif
-#endif
-
 RZ_API void *rz_malloc_aligned(size_t size, size_t alignment) {
```
Comment (wargio) [inline librz/util/alloc.c:7]: do we even use these apis? why we need aligned mem?

**Review by wargio — APPROVED** [review]
<3

File: librz/util/alloc.c (hunk 1 above)
Comment (XVilka) [reply librz/util/alloc.c:7]: It's used in `librz/io/p/io_ptrace.c`

**Comment by wargio** [comment]
looks like you broke debugging

**Review by wargio — APPROVED** [review]

**Pull request merged**